lib/auth/keystore/keystore_test.go

```diff
@@ -338,3 +342,24 @@ func TestKeyStore(t *testing.T) {
 		})
 	}
 }
+
+func TestLicenseRequirement(t *testing.T) {
+	if os.Getenv("SOFTHSM2_PATH") == "" {
+		t.SkipNow()
+	}
+
+	config := keystore.SetupSoftHSMTest(t)
+	config.HostUUID = "server1"
+
+	// should fail to create the keystore with default modules
+	_, err := keystore.NewKeyStore(config)
+	require.Error(t, err)
+
+	defaultModules := modules.GetModules()
+	defer modules.SetModules(defaultModules)
+	modules.SetModules(keystore.TestModules{})
+
+	// should succeed when HSM feature is enabled
+	_, err = keystore.NewKeyStore(config)
```

Comment:
Is there a corresponding "should fail when not enabled" test?

Reply:
The case above with the default modules covers the "not enabled" case